Source/Core/VideoBackends/D3D12/D3DUtil.cpp

```diff
@@ -0,0 +1,882 @@
+	"	float2 tex : TEXCOORD;\n"
+	"};\n"
+	"float4 main( PS_INPUT input ) : SV_Target\n"
+	"{\n"
+	"	return tex2D.Sample( linearSampler, input.tex ) * input.col;\n"
+	"};\n"
+};
+
+const char fontvertshader[] = {
+	"struct VS_INPUT\n"
+	"{\n"
+	"	float4 pos : POSITION;\n"
+	"	float4 col : COLOR;\n"
+	"	float2 tex : TEXCOORD;\n"
+	"};\n"
+	"struct PS_INPUT\n"
+	"{\n"
+	"	float4 pos : SV_POSITION;\n"
+	"	float4 col : COLOR;\n"
+	"	float2 tex : TEXCOORD;\n"
+	"};\n"
+	"PS_INPUT main( VS_INPUT input )\n"
+	"{\n"
+	"	PS_INPUT output;\n"
+	"	output.pos = input.pos;\n"
+	"	output.col = input.col;\n"
+	"	output.tex = input.tex;\n"
+	"	return output;\n"
+	"};\n"
+};
+
+int CD3DFont::Init()
+{
+	// Create vertex buffer for the letters
+
+	// Prepare to create a bitmap
+	unsigned int* pBitmapBits;
+	BITMAPINFO bmi;
+	ZeroMemory(&bmi.bmiHeader, sizeof(BITMAPINFOHEADER));
+	bmi.bmiHeader.biSize        = sizeof(BITMAPINFOHEADER);
+	bmi.bmiHeader.biWidth       =  (int)m_dwTexWidth;
+	bmi.bmiHeader.biHeight      = -(int)m_dwTexHeight;
+	bmi.bmiHeader.biPlanes      = 1;
+	bmi.bmiHeader.biCompression = BI_RGB;
+	bmi.bmiHeader.biBitCount    = 32;
+
+	// Create a DC and a bitmap for the font
+	HDC hDC = CreateCompatibleDC(nullptr);
+	HBITMAP hbmBitmap = CreateDIBSection(hDC, &bmi, DIB_RGB_COLORS, (void**)&pBitmapBits, nullptr, 0);
+	SetMapMode(hDC, MM_TEXT);
+
+	// create a GDI font
+	HFONT hFont = CreateFont(24, 0, 0, 0, FW_NORMAL, FALSE,
+							FALSE, FALSE, DEFAULT_CHARSET, OUT_DEFAULT_PRECIS,
+							CLIP_DEFAULT_PRECIS, PROOF_QUALITY,
+							VARIABLE_PITCH, _T("Tahoma"));
+	if (nullptr == hFont) return E_FAIL;
+
+	HGDIOBJ hOldbmBitmap = SelectObject(hDC, hbmBitmap);
+	HGDIOBJ hOldFont = SelectObject(hDC, hFont);
+
+	// Set text properties
+	SetTextColor(hDC, 0xFFFFFF);
+	SetBkColor  (hDC, 0);
+	SetTextAlign(hDC, TA_TOP);
+
+	TEXTMETRICW tm;
+	GetTextMetricsW(hDC, &tm);
+	m_LineHeight = tm.tmHeight;
+
+	// Loop through all printable characters and output them to the bitmap
+	// Meanwhile, keep track of the corresponding tex coords for each character.
+	int x = 0, y = 0;
+	char str[2] = "\0";
+	for (int c = 0; c < 127 - 32; c++)
+	{
+		str[0] = c + 32;
+		SIZE size;
+		GetTextExtentPoint32A(hDC, str, 1, &size);
+		if ((int)(x+size.cx+1) > m_dwTexWidth)
+		{
+			x  = 0;
+			y += m_LineHeight;
+		}
+
+		ExtTextOutA(hDC, x+1, y+0, ETO_OPAQUE | ETO_CLIPPED, nullptr, str, 1, nullptr);
+		m_fTexCoords[c][0] = ((float)(x+0))/m_dwTexWidth;
+		m_fTexCoords[c][1] = ((float)(y+0))/m_dwTexHeight;
+		m_fTexCoords[c][2] = ((float)(x+0+size.cx))/m_dwTexWidth;
+		m_fTexCoords[c][3] = ((float)(y+0+size.cy))/m_dwTexHeight;
+
+		x += size.cx + 3;  // 3 to work around annoying ij conflict (part of the j ends up with the i)
+	}
+
+	// Create a new texture for the font
+	// possible optimization: store the converted data in a buffer and fill the texture on creation.
+	// That way, we can use a static texture
+	void* texInitialData = malloc(m_dwTexWidth * m_dwTexHeight * 4); // 32 bpp
```

Comment:
You also use `delete` for this a little bit down, and continue with more code after it which indicates you should be moving all of this behavior to its own function (also using delete on malloc'ed data is undefined behavior).

Consider using a unique_ptr for this data instead.